src/perl/vdotest/VDOTest/VDOClobber.pm

```diff
@@ -0,0 +1,121 @@
+##
+# Test that creating a vdo doesn't clobber a file system or another VDO volume
+#
+# $Id: $
+##
+package VDOTest::VDOClobber;
+
+use strict;
+use warnings FATAL => qw(all);
+use English qw(-no_match_vars);
+use Log::Log4perl;
+
+use Permabit::Assertions qw(
+  assertEvalErrorMatches
+  assertNumArgs
+  assertRegexpMatches
+);
+use Permabit::CommandString::VDOFormat;
+use Permabit::Constants;
+use Permabit::PlatformUtils qw(getDistroInfo);
+
+use base qw(VDOTest);
+
+my $log = Log::Log4perl->get_logger(__PACKAGE__);
+
+##
+# @paramList{getProperties}
+our %PROPERTIES
+  = (
+     # @ple set up a Linear device
+     deviceType => "vdo",
+    );
+##
+
+#############################################################################
+##
+sub checkClobberOnlyIfForce {
+  my ($self) = assertNumArgs(1, @_);
+  my $device = $self->getDevice();
+  my $storage = $device->{storageDevice};
+
+  # Try running vdoformat without force. Should complain about signatures.
+  eval {
+    $device->formatVDO({ force => undef });
+  };
+  assertEvalErrorMatches(qr/existing signature|already containing/);
+  # Now create a new VDO using vdoformat. This should work.
+  $device->{_formatted} = 0;
+  $device->start();
+  # Wait till start finishes then stop the device and wipe the geometry block.
+  $device->getMachine->runSystemCmd("sudo udevadm settle");
+  $device->stop();
+  $storage->ddWrite((bs => 4096, count => 1, if => "/dev/zero"));
+}
+
+#############################################################################
+# Test that creating a vdo won't scribble over various types of existing
+# content, unless --force is given to vdoformat.
+##
+sub testClobber {
+  my ($self) = assertNumArgs(1, @_);
+  my $device = $self->getDevice();
+  my $storage = $device->{storageDevice};
+  my $storagePath = $storage->getDevicePath();
+  my $machine = $device->getMachine();
+
+  $device->stop();
+  $self->checkClobberOnlyIfForce();
+
+  my @filesystems = qw(ext3 vfat xfs);
+  # RHEL8 does not support btrfs.
+  if (getDistroInfo($machine->getName()) ne "RHEL8") {
+    push(@filesystems, "btrfs");
+  }
```

Comment:
This tree does not work on RHEL8 and we're not going to backport it that far. If you want to use btrfs, just make it part of the normal list.

Reply:
Removed RHEL8